src/components/modal/modal.tsx

```diff
@@ -34,6 +40,8 @@ export const Modal: FC<ModalProps> = ({ isOpen, onModalClose, title, className,
 
   if (!isOpen) return null
 
+  console.warn('Modal component will be removed in future major version. Use `Dialog` component instead')
```

Comment:
**todo:** Logging content to the browser's console is a side-effect of rendering, so it should be inside a `useEffect`.